rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java

```diff
@@ -254,6 +254,14 @@ public static String generateId(String sourceName, String hostName) {
             return sourceName + "-host-" + hostName;
         }
 
+        public String getIcon() {
+            if (this.isVirtual()) {
+                return RMImages.instance.host_virtual_16().getSafeUri().asString();
+            } else {
+                return RMImages.instance.host_16().getSafeUri().asString();
+            }
+        }
+
```

Comment:
We can change it like this (only one line) :
return this.isVirtual ? RMImages.instance.host_virtual_16().getSafeUri().asString()  : RMImages.instance.host_16().getSafeUri().asString()

Reply:
Applied.